File: prototype/rails/src/app/domain/orders/purchaser.rb
module Domain
  module Orders
    # Who is buying, as much of them as checkout needs. A guest arrives with no
    # verified email and pays after verifying.
    Purchaser = Data.define(:id, :email, :email_verified_at) do
      def email_verified?
        !email_verified_at.nil?
      end
    end
  end
end

File: prototype/node/src/app/core/orders/purchaser.ts
import type { CustomerId } from '../ids/entity-ids.ts'

/**
 * Who is buying, as much of them as checkout needs. A guest arrives with no
 * verified address and pays after verifying one.
 */
export type Purchaser = {
  id: CustomerId
  email: string | null
  isEmailVerified: boolean
}
